llvm/include/llvm/Transforms/Utils/LoopUtils.h

```diff
@@ -318,9 +318,8 @@ void addStringMetadataToLoop(Loop *TheLoop, const char *MDString,
 /// Returns a loop's estimated trip count based on branch weight metadata.
 /// In addition if \p EstimatedLoopInvocationWeight is not null it is
 /// initialized with weight of loop's latch leading to the exit.
-/// Returns a valid positive trip count, or std::nullopt when a meaningful
-/// estimate cannot be made (including when the trip count wouldn't fit in the
-/// result type).
+/// Returns a valid positive trip count, saturated at UINT_MAX, or std::nullopt
```

Comment:
Might be better to clarify that this doesn't aways return a valid trip count, but a lower bound on the trip count?

Reply:
I prefer the current wording actually: the function name already says "estimated", and we do return a exact computation from what's in the metadata, except in the case of overflows, so I think the clarification of saturating at UINT_MAX is more precise.

Also, about the tile: I don't think it's critical to differentiate signed-versus-unsigned int in the title; it's there in the commit message anyway. Would have been nice to do nevertheless, but it's too late, since the patch has already landed.